contracts/Aave.sol

```diff
@@ -78,21 +82,37 @@ contract Aave {
         token.withdraw(amt);
     }
 
-    function deposit(address tokenaddr) external payable {
-        // address tokenaddr = weth;
-        bool isEth = (tokenaddr == eth) ? true : false;
-        tokenaddr = (isEth) ? weth : tokenaddr;
+    function ERCdeposit(address tokenaddr, uint256 amt) external payable {
+        console.log(address(this));
+        IAave aave = IAave(provider.getLendingPool());
+        IERC20 token = IERC20(tokenaddr);
+        require(token.balanceOf(msg.sender) >= amt, "Insufficent funds");
+        console.log(token.balanceOf(msg.sender) / (10**18));
+        console.log(msg.sender);
+        token.approve(address(this), amt);
+        // token.allowance(msg.sender, address(this));
+        token.transferFrom(msg.sender, address(this), amt);
+        token.approve(address(aave), amt);
+        aave.deposit(address(tokenaddr), amt, address(this), 0);
```

Comment:
You can't approve your own address for some amount!
You have to approve this in the script/test